### ISSUE TITLE
Wizard + Gradle + Java 9.0.1 = Boom

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-core#522]
```
java.lang.reflect.InvocationTargetException
	at org.eclipse.xtext.ui.wizard.XtextNewProjectWizard$1.run(XtextNewProjectWizard.java:58)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.lang.IllegalArgumentException: Could not determine java version from '9.0.1'.
	at org.gradle.api.JavaVersion.toVersion(JavaVersion.java:70)
	at org.gradle.api.JavaVersion.current(JavaVersion.java:80)
	at org.gradle.internal.jvm.UnsupportedJavaRuntimeException.assertUsingVersion(UnsupportedJavaRuntimeException.java:29)
	at org.gradle.tooling.internal.consumer.ConnectorServices.checkJavaVersion(ConnectorServices.java:57)
	at org.gradle.tooling.internal.consumer.ConnectorServices.createCancellationTokenSource(ConnectorServices.java:39)
	at org.gradle.tooling.GradleConnector.newCancellationTokenSource(GradleConnector.java:95)
	at org.eclipse.buildship.core.util.progress.ToolingApiJob.<init>(ToolingApiJob.java:57)
	at org.eclipse.buildship.core.workspace.internal.SynchronizeGradleBuildsJob.<init>(SynchronizeGradleBuildsJob.java:46)
	at org.eclipse.buildship.core.workspace.internal.SynchronizeGradleBuildsJob.forSingleGradleBuild(SynchronizeGradleBuildsJob.java:122)
	at org.eclipse.buildship.core.workspace.internal.DefaultGradleBuild.synchronize(DefaultGradleBuild.java:58)
	at org.eclipse.buildship.core.workspace.internal.DefaultGradleBuild.synchronize(DefaultGradleBuild.java:53)
	at org.eclipse.buildship.core.workspace.internal.DefaultGradleBuild.synchronize(DefaultGradleBuild.java:48)
	at org.eclipse.buildship.core.workspace.internal.GradleClasspathContainerInitializer.loadClasspath(GradleClasspathContainerInitializer.java:58)
	at org.eclipse.buildship.core.workspace.internal.GradleClasspathContainerInitializer.initialize(GradleClasspathContainerInitializer.java:39)
	at org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer(JavaModelManager.java:3112)
	at org.eclipse.jdt.internal.core.JavaModelManager$10.run(JavaModelManager.java:3018)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.jdt.internal.core.JavaModelManager.initializeAllContainers(JavaModelManager.java:3058)
	at org.eclipse.jdt.internal.core.JavaModelManager.getClasspathContainer(JavaModelManager.java:2056)
	at org.eclipse.jdt.core.JavaCore.getClasspathContainer(JavaCore.java:3548)
	at org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:2986)
	at org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:2909)
	at org.eclipse.jdt.internal.core.ClasspathEntry.validateClasspath(ClasspathEntry.java:1826)
	at org.eclipse.jdt.internal.core.SetClasspathOperation.verify(SetClasspathOperation.java:118)
	at org.eclipse.jdt.internal.core.JavaModelOperation.runOperation(JavaModelOperation.java:784)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3390)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3352)
	at org.eclipse.jdt.internal.core.JavaProject.setRawClasspath(JavaProject.java:3405)
	at org.eclipse.xtext.ui.util.JavaProjectFactory.enhanceProject(JavaProjectFactory.java:71)
	at org.eclipse.xtext.ui.util.ProjectFactory.createProject(ProjectFactory.java:167)
	at org.eclipse.xtext.xtext.ui.wizard.project.XtextProjectCreator.createJavaProject(XtextProjectCreator.java:108)
	at org.eclipse.xtext.xtext.ui.wizard.project.XtextProjectCreator.createProject(XtextProjectCreator.java:87)
	at org.eclipse.xtext.xtext.ui.wizard.project.XtextProjectCreator.execute(XtextProjectCreator.java:71)
	at org.eclipse.ui.actions.WorkspaceModifyOperation.lambda$0(WorkspaceModifyOperation.java:107)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.ui.actions.WorkspaceModifyOperation.run(WorkspaceModifyOperation.java:128)
	at org.eclipse.xtext.ui.wizard.XtextNewProjectWizard.doFinish(XtextNewProjectWizard.java:83)
	at org.eclipse.xtext.ui.wizard.XtextNewProjectWizard$1.run(XtextNewProjectWizard.java:55)
	... 1 more

```

i assume going to gradle `4.2.1` or later in the wizard helps

**Comment by cdietrich** [comment]
nope that does not seem to help
=> maybe newer buildship version needed too ?!?

**Comment by cdietrich** [comment]
https://github.com/eclipse/buildship/issues/601

**Comment by cdietrich** [comment]
using the latest buildship and gralde 4.4 seems to help.
but i found this one when really using java 9
https://github.com/xtext/xtext-gradle-plugin/issues/101

**Comment by cdietrich** [comment]
increased version created by the wizard to 4.4
merge of fix to xtext gradle plugin and updating the version of the plugin is still outstanding.
same for the version increase of asm in maven/gradle

**Comment by cdietrich** [comment]
https://github.com/eclipse/xtext-extras/issues/211

**Comment by cdietrich** [comment]
Fixed in 2.14
